scripts/zkai_attention_kv_multihead_quantized_softmax_receipt_gate.py

```
@@ -0,0 +1,782 @@
+    per_head_step_row_counts: list[dict[str, int]] = []
+    max_rows_per_head_step = 0
+    denominator_sum = 0
+    for head_step in sorted(grouped):
+        head_index, step_index = head_step
+        step_rows = grouped[head_step]
+        if not step_rows:
+            raise MultiheadQuantizedSoftmaxReceiptGateError(f"{profile.profile_id} empty head/step rows")
+        output_index = output_map[head_step]
+        selected_scores = {row["selected_score"] for row in step_rows}
+        if len(selected_scores) != 1:
+            raise MultiheadQuantizedSoftmaxReceiptGateError(f"{profile.profile_id} selected score drift")
```

Ruff (0.15.12)

[warning] Avoid specifying long messages outside the exception class

(TRY003)